source/contribute.rst

```diff
@@ -192,6 +195,70 @@ this in case you want to read more about this:
 - `Commit message style for git <https://commit.style/>`_
 - `A note about git commit messages <https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html>`_
 
+To squash or not to squash
+--------------------------
+
+Of course, if you want to craft good commit messages, you will have a
+hard time if the changeset you are describing does too many things. That
+might very well happen if you notice small things along the way that are
+unrelated to your PR, but too small to warrant a separate one. ``git add
+--patch`` or ``git add -p`` will be of invaluable help to commit things
```

Comment:
I think before going this deep, in the very first paragraph, we should give the contributor the right to not deal with all our perfectionism and just squash everything in one commit. Most of the developers making tiny patches will be happy right after this point and won't have to read the rest.